staging/src/k8s.io/component-base/traces/utils.go

```diff
@@ -26,33 +26,64 @@ import (
 	"go.opentelemetry.io/otel/propagation"
 	"go.opentelemetry.io/otel/sdk/resource"
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/semconv"
 	"go.opentelemetry.io/otel/trace"
 
 	"k8s.io/client-go/transport"
-	"k8s.io/klog/v2"
 )
 
-// NewProvider initializes tracing in the component, and enforces recommended tracing behavior.
-func NewProvider(ctx context.Context, baseSampler sdktrace.Sampler, resourceOpts []resource.Option, opts ...otlpgrpc.Option) trace.TracerProvider {
-	opts = append(opts, otlpgrpc.WithInsecure())
+// InitTracing initialized tracing in the component, and enforces recommended tracing behavior
+func InitTracing(ctx context.Context,
+	traceID string,
+	serviceName string,
+	samplingRate *int32,
+	opts []otlpgrpc.Option,
+) (trace.TracerProvider, error) {
+	resourceOpts := []resource.Option{
```

Comment:
I realize the resource code is duplicated in components, but i'd still prefer passing in resourceOpts as a parameter so that components can easily add resource attributes if they want in the future.

Reply:
makes sense, ok